docs/docs/guides/smart_contracts/index.md

```diff
@@ -3,3 +3,346 @@ sidebar_position: 4
 ---
 
 # Smart Contracts
+
+:::info
+This guide expects you to have some basic knowledge. If you are just starting, it is recommended to first check out this [Tutorial: Deploying and Interacting with Smart Contracts](./deploying_and_interacting_with_smart_contracts).
+:::
+
+## Contract class
+
+The `Contract` class is the main object exported by the `web3-eth-contract` package. It is also available in the `web3` package.
+
+### Importing the Contract Class 
+
+To use the `Contract` class, you'll need to import it from one of two packages: directly from the `web3-eth-contract` package or from the `web3` package.
+
+Here's an example of importing from each:
+
+```ts
+// Importing from web3-eth-contract package
+import { Contract } from 'web3-eth-contract';
+const contract = new Contract(...);
+
+// Importing from the main web3 package
+import { Contract } from 'web3';
+const contract = new Contract(...);
+
+// Importing from the main web3 package from inside `Web3.eth` namespace
+import { Web3 } from 'web3';
+const contract = new Web3.eth.Contract(...);
+
+// to set the provider for the contract instance:
+contract.setProvider('yourProvider');
+```
+
+### `Contract` vs `web3.eth.Contract`
+
+There is a way to create a contract object, other than the ones listed above. That is by accessing an instance of `Web3` object. And this instance of `Web3` is usually called `web3`.
+
+Actually, the `web3.eth.Contract` is typically how you access the class through a web3 instance that already has a provider setup and that may already has customized configurations.
+
+:::tip
+Note the difference between `Web3.eth.Contract` and `web3instance.eth.Contract` (that is usually named `web3.eth.Contract`) is that the `Web3` (with capital `W`) is used to access the objects organized in namespaces. While `web3instance`, that is usually named `web3`, is to access the properties of this instance. 
+:::
+
+Examples:
+
+```ts
+import { Web3 } from 'web3';
+
+// Instantiating Contract directly with provider URL
+const contract = new Web3.eth.Contract(abi, address, { provider: 'ws://localhost:8546' }); // not the capital 'W' in Web3 here
```

Comment:
without creating Web3 instance following line will error: 
import { Web3 } from 'web3';
new Web3.eth.Contract(abi, address, { provider: 'ws://localhost:8546' });

in Web3.eth  eth is undefined, 
did you check all code examples in this PR?